EXT4: fix online resize bug

From: Josef Bacik <jbacik@redhat.com>

There is a bug when we are trying to verify that the reserve inode's double
indirect blocks point back to the primary gdt blocks.  The fix is obvious, we
need to mod the gdb count by the addr's per block.  This was verified using the
same testcase as with the ext3 equivalent of this patch.  Thank you,

Signed-off-by: Josef Bacik <jbacik@redhat.com>
Signed-off-by: Mingming Cao <cmm@us.ibm.com>


---
 fs/ext4/resize.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-2.6.26-rc4/fs/ext4/resize.c
===================================================================
--- linux-2.6.26-rc4.orig/fs/ext4/resize.c	2008-05-26 11:08:11.000000000 -0700
+++ linux-2.6.26-rc4/fs/ext4/resize.c	2008-05-30 11:05:39.000000000 -0700
@@ -563,7 +563,8 @@ static int reserve_backup_gdb(handle_t *
 	}
 
 	blk = EXT4_SB(sb)->s_sbh->b_blocknr + 1 + EXT4_SB(sb)->s_gdb_count;
-	data = (__le32 *)dind->b_data + EXT4_SB(sb)->s_gdb_count;
+	data = (__le32 *)dind->b_data + (EXT4_SB(sb)->s_gdb_count %
+					 EXT4_ADDR_PER_BLOCK(sb));
 	end = (__le32 *)dind->b_data + EXT4_ADDR_PER_BLOCK(sb);
 
 	/* Get each reserved primary GDT block and verify it holds backups */
